### DISCUSSION TITLE
Welcome to monaco-react Discussions!

### DISCUSSION THREAD

**Discussion by suren-atoyan** [suren-atoyan/monaco-react#144] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about
  * Share ideas
  * Engage with other community members
  * Welcome others and are open-minded

**Comment by sunnyday90210** [comment]
Hi,
Was wondering if anyone has ever gotten Rich IntelliSense Validation for python to work? I have not been able to find very many resources!

**Comment by suren-atoyan** [comment]
Hello @sunnyday90210 

Unfortunately `monaco-editor` doesn't support rich `IntelliSense` and validation for `Python` out of the box. Maybe there are external libraries that can help you to achieve that, but from out of the box there is no rich `IntelliSense` and validation for `Python`.

The whole supported languages are divided into two groups:

1) languages that have rich `IntelliSense` and validation

- `TypeScript`
- `JavaScript`
- `CSS`
- `LESS`
- `SCSS`
- `JSON`
- `HTML`

2) languages with only basic syntax colorization

- `XML`
- `PHP`
- `C#`
- `C++`
- `Razor`
- `Markdown`
- `Diff`
- `Java`
- `VB`
- `CoffeeScript`
- `Handlebars`
- `Batch`
- `Pug`
- `F#`
- `Lua`
- `Powershell`
- `Python`
- `Ruby`
- `SASS`
- `R`
- `Objective-C`

Unfortunately, `Python` is from the second group.
If you find an external library to achieve that, please let us know here. `Python` is a great language, and it would be great to have it work as expected in browser-based editors.

**Comment by suren-atoyan** [comment]
[Here](https://mono.software/2017/04/11/custom-intellisense-with-monaco-editor/) is the article about custom `IntelliSense` for `XML`

**Comment by suren-atoyan** [comment]
[Here](https://github.com/microsoft/monaco-editor/issues/421) is an interesting discussion about that, with useful links

**Reply by sunnyday90210** [reply]
Thank you for the quick response and resources, I will check these out and let you know if we make any progress 🙏
